src/core/providers/strategies/LedgerProviderStrategy/helpers/authenticateLedgerAccount.ts

```diff
@@ -19,154 +19,160 @@ type GetLedgerLoginType = {
     token?: string;
   };
   config: LedgerConfigType;
-  manager: LedgerConnectStateManagerType | null;
+  manager: LedgerConnectStateManager | null;
   provider: HWProvider | null;
-  eventBus?: LedgerEventBusType | null;
+  eventBus?: IEventBus | null;
   login: LedgerLoginType | null;
 };
 
-export const authenticateLedgerAccount = async ({
+export function authenticateLedgerAccount({
   options,
   config,
   manager,
   provider,
   eventBus,
   login
-}: GetLedgerLoginType) => {
+}: GetLedgerLoginType) {
   const authData = getAuthTokenText({
     loginToken: options?.token,
     version: config.version
   });
 
   // refresh account list
-  await updateAccountsList({ manager, provider });
-
-  // cycle trough accounts until user makes a choice
-  const selectedAccount = await new Promise<{
-    address: string;
-    signature: string;
-    addressIndex: number;
-  }>(async (resolve, reject) => {
-    const closeComponent = () => {
-      manager?.closeAndReset();
-    };
+  return updateAccountsList({ manager, provider }).then(async function () {
+    // cycle trough accounts until user makes a choice
+    const selectedAccount = await new Promise<{
+      address: string;
+      signature: string;
+      addressIndex: number;
+    }>(async function (resolve, reject) {
+      function closeComponent() {
+        manager?.closeAndReset();
+      }
 
-    const onNextPageChanged = async () => {
-      const startIndex = manager?.getAccountScreenData()?.startIndex || 0;
-      manager?.updateStartIndex(startIndex + manager.addressesPerPage);
-      await updateAccountsList({ manager, provider });
-    };
+      async function onNextPageChanged() {
```

Comment:
Same comment as above, applicable to the instances below, the function should be prefixed by `handle` instead of `on`.